src/components/layout/tables/AntRow/AntRow.tsx

```diff
@@ -0,0 +1,183 @@
+  const [targetId, setTargetId] = useState<string>();
+
+  const dataProvider = defaultDataProvider(arweave);
+
+  // todo: implement error antState for row items
+  useEffect(() => {
+    getStatus();
+    loadAntState();
+  }, [antId]);
+
+  useEffect(() => {
+    if (antState) {
+      getName();
+      getTargetId();
+    }
+  }, [antState]);
+
+  async function loadAntState() {
+    const state = await dataProvider.getContractState(antId);
+    setAntState(state);
+    return;
+  }
+
+  async function getStatus() {
+    const confirmations = await dataProvider.getContractConfirmations(antId);
+    setConfirmations(confirmations);
+
+    if (confirmations > 0 && confirmations < 50) {
+      setStatusIcon(
+        <AlertTriangleIcon width={20} height={20} fill={'var(--accent)'} />,
+      );
+    }
+    if (confirmations > 49) {
+      setStatusIcon(
+        <CircleCheck width={20} height={20} fill={'var(--success-green)'} />,
+      );
+    }
+    if (confirmations <= 0) {
+      setStatusIcon(
+        <AlertCircle width={20} height={20} fill={'var(--text-faded)'} />,
+      );
+    }
+  }
+
+  function getName() {
+    if (!antState) {
+      return;
+    }
+    if (antState.name.length > 20) {
+      setNickname(
+        `${antState.name.slice(0, 10)}...${antState.name.slice(-10)}`,
+      );
+    }
+    if (antState.name.length < 20) {
+      setNickname(antState.name);
+    }
+  }
+  function getTargetId() {
+    if (antState.records['@'].transactionId) {
+      setTargetId(antState.records['@'].transaction);
+    }
+    // todo: remove below for v1, this is a legacy format for ant record antStates.
+    if (typeof antState.records['@'] === 'string') {
+      setTargetId(antState.records['@']);
+    }
+  }
+
+  return (
+    <>
+      <tr
+        className="assets-table-header"
+        style={
+          bgColor
+            ? {
+                backgroundColor: bgColor,
+                color: textColor,
+              }
+            : {}
+        }
+      >
+        <>
+          <td
+            className="assets-table-item"
+            style={textColor ? { color: textColor } : {}}
+          >
+            {nickname ? nickname : <Loader size={30} />}
+          </td>
+          <td
+            className="assets-table-item"
+            style={textColor ? { color: textColor } : {}}
+          >
+            {isMobile ? (
+              <CopyTextButton
+                displayText={`${antId.slice(0, 2)}...${antId.slice(-2)}`}
+                copyText={antId}
+                size={24}
+              />
+            ) : (
+              <CopyTextButton
+                displayText={`${antId.slice(0, 6)}...${antId.slice(-6)}`}
+                copyText={antId}
+                size={24}
+              />
+            )}
+          </td>
+          <td
+            className="assets-table-item"
+            style={textColor ? { color: textColor } : {}}
+          >
+            {typeof targetId == 'string' ? (
```

Comment:
why is this check needed?

Reply:
oh, its not. I was setting the conditional `<Loader />` comps in the detail state at first as the initial state, and i was having trouble setting the conditional to change rendering the loader, and thats what i came up with - but since I changed that is simply now `{targetId ? ...jsx...}`